src/main/scala/coreplex/Coreplex.scala

```diff
@@ -65,21 +80,82 @@ abstract class Coreplex(implicit val p: Parameters, implicit val c: CoreplexConf
 }
 
 class DefaultCoreplex(tp: Parameters, tc: CoreplexConfig) extends Coreplex()(tp, tc) {
+  class TileWrapper(tileBuilder: (Bool, Parameters) => Tile, _clock: Clock, _reset: Bool)
```

Comment:
And if did this in a subclass, we would still need the wrapper. What it's doing is changing the clock domain (using _clock and _reset), then putting in synchronizers from the tile clock domain to the external clock domain (ext_clock and ext_reset). I think that's the cleanest way of doing it.